packages/core/test/final.test.ts

```diff
@@ -231,4 +231,256 @@ describe('final states', () => {
 
     expect(spy).toHaveBeenCalledWith(1);
   });
+
+  it('should emit a done state event for a parallel state when its parallel children reach their final states', () => {
```

Comment:
Just for my own reference
![CleanShot 2023-10-16 at 13 44 24](https://github.com/statelyai/xstate/assets/1093738/1933b59a-f681-4407-adc0-a81589eda58e)